legacy/python/platform/src/actions/install_deps.rs

```diff
@@ -0,0 +1,129 @@
+use moon_action::Operation;
+use moon_console::{Checkpoint, Console};
+use moon_python_tool::PythonTool;
+use moon_utils::get_workspace_root;
+use std::path::Path;
+
+use crate::find_requirements_txt;
+
+pub async fn install_deps(
+    python: &PythonTool,
+    working_dir: &Path,
+    console: &Console,
+) -> miette::Result<Vec<Operation>> {
+    let mut operations = vec![];
+
+    // python.exec_python(args, working_dir)
+
+    if let Some(pip_config) = &python.config.pip {
+        // Very first step: Activate virtual environment
+        console
+            .out
+            .print_checkpoint(Checkpoint::Setup, format!("activate virtual environment"))?;
+        let virtual_environment = &get_workspace_root().join(python.config.venv_name.clone());
+
+        if !virtual_environment.exists() {
+            let args = vec![
+                "-m",
+                "venv",
+                virtual_environment.as_os_str().to_str().unwrap(),
+            ];
+            operations.push(
+                Operation::task_execution(format!("python {} ", args.join(" ")))
+                    .track_async(|| python.exec_python(args, working_dir))
+                    .await?,
+            );
+        }
+
+        if let Some(pip_version) = &pip_config.version {
```

Comment:
This shouldn't be done here. Use the setup toolchain action instead.

Reply:
Is this not the same like bins in the rust toolchain?
Because also here it will be executed in the install_deps action.

Reply:
I have had a look into this topic. From what I understand is that the function setup_toolchain is used for installing python via proto onto the system (~/.proto).

I have implemented now the virtual environment into the setup_tools function, because this will also be called when doing a moon sync command.